pkg/stream/xprotocol/conn.go

```diff
@@ -153,6 +157,14 @@ func (sc *streamConn) Dispatch(buf types.IoBuffer) {
 		// 2.4 prepare next
 		sc.ctxManager.Next()
 	}
+
+	sc.checkGracefulShutdown()
+}
+
+func (sc *streamConn) checkGracefulShutdown() {
```

Comment:
The `streamConn` maybe point to two types, `types.ClientStreamConnection` or `types.ServerStreamConnection` which need to be handled differently.

Reply:
Thanks, addressed.